Library/Homebrew/dev-cmd/audit.rb

```diff
@@ -236,8 +236,29 @@ def self.check_http_content(url, name, user_agents: [:default])
       details[:content_length] == secure_details[:content_length]
     file_match = details[:file_hash] == secure_details[:file_hash]
 
-    return if !etag_match && !content_length_match && !file_match
-    "The URL #{url} could use HTTPS rather than HTTP"
+    if etag_match || content_length_match || file_match
+      return "The URL #{url} could use HTTPS rather than HTTP"
+    end
+
+    return if type != "homepage"
+
+    details[:file] = details[:file].gsub(%r{https?:\\?\/\\?\/}, "/")
+    secure_details[:file] = secure_details[:file].gsub(%r{https?:\\?\/\\?\/}, "/")
+
+    # Same content after normalization
+    if details[:file] == secure_details[:file]
+      return "The URL #{url} could use HTTPS rather than HTTP"
+    end
+
+    # Same size, different content after normalization
+    # (typical causes: Generated ID, Timestamp, Unix time)
+    if details[:file].length == secure_details[:file].length
+      return "The URL #{url} could use HTTPS rather than HTTP"
```

Comment:
This message I think should be the same as below; it may be able to use it but should be verified.